core/node/state_keeper/src/io/mempool.rs

```diff
@@ -298,6 +299,23 @@ impl StateKeeperIO for MempoolIO {
                         .await?;
                     continue;
                 }
+
+                // Reject transactions that violate block.timestamp constraints. Such transactions should be
+                // rejected at the API level, but we need to protect ourselves in case if a transaction
+                // goes outside of the allowed range while being in the mempool
+                let matches_range = constraint.timestamp_asserter_range.map_or(true, |x| {
+                    x.0 < l2_block_timestamp as i64 && x.1 > l2_block_timestamp as i64
```

Comment:
Bikeshedding: IMO, a range with both exclusive ends is somewhat esoteric, even not taking Rust into account (there's no easy way to construct such ranges, unlike `x..y` / `x..=y`). If I'd read the `assertTimestampInRange` name, I'd assume that both range ends are inclusive. I understand that this is very subjective though.

Reply:
Refined the logic with `range.contains` instead